In [3]:
# import the libraries
import pandas as pd
import numpy as np

## Task 1

In [4]:
# path where data is located
data_source_path = 'https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M'

In [5]:
# read the webpage into a datafrome
df_list = pd.read_html(data_source_path, header=None)

## Task 2

The dataframe will consist of three columns: PostalCode, Borough, and Neighborhood

In [6]:
# store the data in dataframe from first row of the list
df = df_list[0]
df.head()

Postcode           Borough     Neighbourhood
0      M1A      Not assigned      Not assigned
1      M2A      Not assigned      Not assigned
2      M3A        North York         Parkwoods
3      M4A        North York  Victoria Village
4      M5A  Downtown Toronto      Harbourfront

Only process the cells that have an assigned borough. Ignore cells with a borough that is Not assigned.

In [7]:
# filter the data where Borough is 'Not assigned'
df_filtred = df[df['Borough'] !='Not assigned']
df_filtred.head()
#df_filtred.shape

Postcode           Borough     Neighbourhood
2      M3A        North York         Parkwoods
3      M4A        North York  Victoria Village
4      M5A  Downtown Toronto      Harbourfront
5      M6A        North York  Lawrence Heights
6      M6A        North York    Lawrence Manor

## Task 3

More than one neighborhood can exist in one postal code area. For example, in the table on the Wikipedia page, you will notice that M5A is listed twice and has two neighborhoods: Harbourfront and Regent Park. These two rows will be combined into one row with the neighborhoods separated with a comma as shown in row 11 in the above table.

In [8]:
df_aggr = df_filtred.groupby(['Postcode', 'Borough'],as_index=False).agg(lambda x :  ', '.join(x))
df_aggr.head(10)

Postcode      Borough                                    Neighbourhood
0      M1B  Scarborough                                   Rouge, Malvern
1      M1C  Scarborough           Highland Creek, Rouge Hill, Port Union
2      M1E  Scarborough                Guildwood, Morningside, West Hill
3      M1G  Scarborough                                           Woburn
4      M1H  Scarborough                                        Cedarbrae
5      M1J  Scarborough                              Scarborough Village
6      M1K  Scarborough      East Birchmount Park, Ionview, Kennedy Park
7      M1L  Scarborough                  Clairlea, Golden Mile, Oakridge
8      M1M  Scarborough  Cliffcrest, Cliffside, Scarborough Village West
9      M1N  Scarborough                      Birch Cliff, Cliffside West

In [9]:
df_aggr.shape

(103, 3)

If a cell has a borough but a Not assigned neighborhood, then the neighborhood will be the same as the borough. So for the 9th cell in the table on the Wikipedia page, the value of the Borough and the Neighborhood columns will be Queen's Park.

In [10]:
df_aggr[df_aggr['Neighbourhood'] =='Not assigned'].head()

Postcode       Borough Neighbourhood
85      M7A  Queen's Park  Not assigned

after the aggrigation there is only one row where Neighbourhood is Not assigned lets replace the value with Borough

In [11]:
 df_aggr['Neighbourhood'] = np.where(df_aggr['Neighbourhood'] == 'Not assigned', df_aggr['Borough'], df_aggr['Neighbourhood'])

In [12]:
df_aggr[df_aggr['Neighbourhood'] =='Not assigned'].head()

Empty DataFrame
Columns: [Postcode, Borough, Neighbourhood]
Index: []

after the replacement there is no values in the 'Neighbourhood' colun with not assigned

In [13]:
df_aggr[df_aggr['Postcode'] =='M7A'].head()

Postcode       Borough Neighbourhood
85      M7A  Queen's Park  Queen's Park

Borough values is assigned to Neighbourhood

In [14]:
df_aggr.shape

(103, 3)

## Part 2 - Getting latitude and longitude

google API  is not working so I took  the csv file and join it with dataframe from previous assignment

In [15]:
geo_path = 'http://cocl.us/Geospatial_data'
# read the csv into a datafrome
df_geolist = pd.read_csv(geo_path)

In [16]:
df_geolist.head()

Postal Code   Latitude  Longitude
0         M1B  43.806686 -79.194353
1         M1C  43.784535 -79.160497
2         M1E  43.763573 -79.188711
3         M1G  43.770992 -79.216917
4         M1H  43.773136 -79.239476

In [17]:
# join the dataframe from privious assignment with latitude and longitude datafrmae
df_latlong = df_aggr.join(df_geolist.set_index('Postal Code'), on='Postcode')
df_latlong.head()

Postcode      Borough                           Neighbourhood   Latitude  \
0      M1B  Scarborough                          Rouge, Malvern  43.806686   
1      M1C  Scarborough  Highland Creek, Rouge Hill, Port Union  43.784535   
2      M1E  Scarborough       Guildwood, Morningside, West Hill  43.763573   
3      M1G  Scarborough                                  Woburn  43.770992   
4      M1H  Scarborough                               Cedarbrae  43.773136   

   Longitude  
0 -79.194353  
1 -79.160497  
2 -79.188711  
3 -79.216917  
4 -79.239476

##    Part 3. Explore Neighborhoods in Toronto

In [18]:
df_toronto = df_latlong[df_latlong['Borough'].str.contains("Toronto")].reset_index(drop=True)
df_toronto.head()

Postcode          Borough                   Neighbourhood   Latitude  \
0      M4E     East Toronto                     The Beaches  43.676357   
1      M4K     East Toronto    The Danforth West, Riverdale  43.679557   
2      M4L     East Toronto  The Beaches West, India Bazaar  43.668999   
3      M4M     East Toronto                 Studio District  43.659526   
4      M4N  Central Toronto                   Lawrence Park  43.728020   

   Longitude  
0 -79.293031  
1 -79.352188  
2 -79.315572  
3 -79.340923  
4 -79.388790

In [19]:
df_toronto.size

190

we have found 190 Borough that contains word Toronto in it

In [20]:
import json # library to handle JSON files

from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

#!conda install -c conda-forge folium=0.5.0 --yes # uncomment this line if you haven't completed the Foursquare API lab
import folium # map rendering library

print('Libraries imported.')

Libraries imported.


In [24]:
latitude = 43.651070
longitude = -79.347015


In [26]:
# create map of toronto using latitude and longitude values
map_toronto = folium.Map(location=[latitude, longitude], zoom_start=11)

# add markers to map
for lat, lng, label in zip(df_toronto['Latitude'], df_toronto['Longitude'], df_toronto['Neighbourhood']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_toronto)  
    
map_toronto

Next, we are going to start utilizing the Foursquare API to explore the neighborhoods and segment them.

#### Define Foursquare Credentials and Version

In [27]:
CLIENT_ID = 'TFBCLECRNLWKLIINFW0ICRZ0WJKOCZNRSQFU1XA2OJCI53G2' # your Foursquare ID
CLIENT_SECRET = 'DLMBSFN3L2IOIGIFDCO5MWKU1CZL4N21OWPYGYSLNQYZTOLX' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: TFBCLECRNLWKLIINFW0ICRZ0WJKOCZNRSQFU1XA2OJCI53G2
CLIENT_SECRET:DLMBSFN3L2IOIGIFDCO5MWKU1CZL4N21OWPYGYSLNQYZTOLX


In [36]:
LIMIT = 100 # limit of number of venues returned by Foursquare API
radius = 500 # define radius

In [87]:
def getNearbyVenues(names, latitudes, longitudes, radius=500):  
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighbourhood', 
                  'Neighbourhood Latitude', 
                  'Neighbourhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [88]:
toronto_venues = getNearbyVenues(names=df_toronto['Neighbourhood'],
                                   latitudes=df_toronto['Latitude'],
                                   longitudes=df_toronto['Longitude']
                                  )

The Beaches
The Danforth West, Riverdale
The Beaches West, India Bazaar
Studio District
Lawrence Park
Davisville North
North Toronto West
Davisville
Moore Park, Summerhill East
Deer Park, Forest Hill SE, Rathnelly, South Hill, Summerhill West
Rosedale
Cabbagetown, St. James Town
Church and Wellesley
Harbourfront
Ryerson, Garden District
St. James Town
Berczy Park
Central Bay Street
Adelaide, King, Richmond
Harbourfront East, Toronto Islands, Union Station
Design Exchange, Toronto Dominion Centre
Commerce Court, Victoria Hotel
Roselawn
Forest Hill North, Forest Hill West
The Annex, North Midtown, Yorkville
Harbord, University of Toronto
Chinatown, Grange Park, Kensington Market
CN Tower, Bathurst Quay, Island airport, Harbourfront West, King and Spadina, Railway Lands, South Niagara
Stn A PO Boxes 25 The Esplanade
First Canadian Place, Underground city
Christie
Dovercourt Village, Dufferin
Little Portugal, Trinity
Brockton, Exhibition Place, Parkdale Village
High Park, The Junction Sout

### Let's check the size of the resulting dataframe

In [89]:
print(toronto_venues.shape)
toronto_venues.head()

(1693, 7)


Neighbourhood  Neighbourhood Latitude  Neighbourhood Longitude  \
0   The Beaches               43.676357               -79.293031   
1   The Beaches               43.676357               -79.293031   
2   The Beaches               43.676357               -79.293031   
3   The Beaches               43.676357               -79.293031   
4   The Beaches               43.676357               -79.293031   

                                Venue  Venue Latitude  Venue Longitude  \
0                   Glen Manor Ravine       43.676821       -79.293942   
1  The Big Carrot Natural Food Market       43.678879       -79.297734   
2                 Grover Pub and Grub       43.679181       -79.297215   
3                 Glen Stewart Ravine       43.676300       -79.294784   
4                       Upper Beaches       43.680563       -79.292869   

         Venue Category  
0                 Trail  
1     Health Food Store  
2                   Pub  
3  Other Great Outdoors  
4          Neighborhood

Let's check how many venues were returned for each neighborhood

In [90]:
toronto_venues.groupby('Neighbourhood').count()

Neighbourhood Latitude  \
Neighbourhood                                                                
Adelaide, King, Richmond                                               100   
Berczy Park                                                             57   
Brockton, Exhibition Place, Parkdale Village                            21   
Business Reply Mail Processing Centre 969 Eastern                       19   
CN Tower, Bathurst Quay, Island airport, Harbou...                      16   
Cabbagetown, St. James Town                                             45   
Central Bay Street                                                      82   
Chinatown, Grange Park, Kensington Market                               95   
Christie                                                                17   
Church and Wellesley                                                    85   
Commerce Court, Victoria Hotel                                         100   
Davisville                                                              38   
Davisville North                                                         7   
Deer Park, Forest Hill SE, Rathnelly, South Hil...                      16   
Design Exchange, Toronto Dominion Centre                               100   
Dovercourt Village, Dufferin                                            14   
First Canadian Place, Underground city                                 100   
Forest Hill North, Forest Hill West                                      4   
Harbord, University of Toronto                                          35   
Harbourfront                                                            51   
Harbourfront East, Toronto Islands, Union Station                      100   
High Park, The Junction South                                           24   
Lawrence Park                                                            3   
Little Portugal, Trinity                                                64   
Moore Park, Summerhill East                                              3   
North Toronto West                                                      21   
Parkdale, Roncesvalles                                                  15   
Rosedale                                                                 4   
Roselawn                                                                 1   
Runnymede, Swansea                                                      35   
Ryerson, Garden District                                               100   
St. James Town                                                         100   
Stn A PO Boxes 25 The Esplanade                                         98   
Studio District                                                         38   
The Annex, North Midtown, Yorkville                                     21   
The Beaches                                                              5   
The Beaches West, India Bazaar                                          17   
The Danforth West, Riverdale                                            42   

                                                    Neighbourhood Longitude  \
Neighbourhood                                                                 
Adelaide, King, Richmond                                                100   
Berczy Park                                                              57   
Brockton, Exhibition Place, Parkdale Village                             21   
Business Reply Mail Processing Centre 969 Eastern                        19   
CN Tower, Bathurst Quay, Island airport, Harbou...                       16   
Cabbagetown, St. James Town                                              45   
Central Bay Street                                                       82   
Chinatown, Grange Park, Kensington Market                                95   
Christie                                                                 17   
Church and Wellesley                                          

Let's find out how many unique categories can be curated from all the returned venues

In [91]:
print('There are {} uniques categories.'.format(len(toronto_venues['Venue Category'].unique())))

There are 232 uniques categories.


### Analyze Each Neighborhood

In [92]:
# one hot encoding
toronto_onehot = pd.get_dummies(toronto_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
toronto_onehot['Neighbourhood'] = toronto_venues['Neighbourhood'] 

# move neighborhood column to the first column
fixed_columns = [toronto_onehot.columns[-1]] + list(toronto_onehot.columns[:-1])
toronto_onehot = toronto_onehot[fixed_columns]

toronto_onehot.head()

Neighbourhood  Afghan Restaurant  Airport  Airport Food Court  Airport Gate  \
0   The Beaches                  0        0                   0             0   
1   The Beaches                  0        0                   0             0   
2   The Beaches                  0        0                   0             0   
3   The Beaches                  0        0                   0             0   
4   The Beaches                  0        0                   0             0   

   Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0               0                0                 0                    0   
1               0                0                 0                    0   
2               0                0                 0                    0   
3               0                0                 0                    0   
4               0                0                 0                    0   

   Antique Shop  ...  Toy / Game Store  Trail  Train Station  \
0             0  ...                 0      1              0   
1             0  ...                 0      0              0   
2             0  ...                 0      0              0   
3             0  ...                 0      0              0   
4             0  ...                 0      0              0   

   Vegetarian / Vegan Restaurant  Video Game Store  Vietnamese Restaurant  \
0                              0                 0                      0   
1                              0                 0                      0   
2                              0                 0                      0   
3                              0                 0                      0   
4                              0                 0                      0   

   Wine Bar  Wine Shop  Wings Joint  Yoga Studio  
0         0          0            0            0  
1         0          0            0            0  
2         0          0            0            0  
3         0          0            0            0  
4         0          0            0            0  

[5 rows x 233 columns]

In [93]:
toronto_grouped = toronto_onehot.groupby('Neighbourhood').mean().reset_index()
toronto_grouped.head()

Neighbourhood  Afghan Restaurant  \
0                           Adelaide, King, Richmond                0.0   
1                                        Berczy Park                0.0   
2       Brockton, Exhibition Place, Parkdale Village                0.0   
3  Business Reply Mail Processing Centre 969 Eastern                0.0   
4  CN Tower, Bathurst Quay, Island airport, Harbo...                0.0   

   Airport  Airport Food Court  Airport Gate  Airport Lounge  Airport Service  \
0   0.0000              0.0000        0.0000           0.000           0.0000   
1   0.0000              0.0000        0.0000           0.000           0.0000   
2   0.0000              0.0000        0.0000           0.000           0.0000   
3   0.0000              0.0000        0.0000           0.000           0.0000   
4   0.0625              0.0625        0.0625           0.125           0.1875   

   Airport Terminal  American Restaurant  Antique Shop  ...  Toy / Game Store  \
0            0.0000                 0.03           0.0  ...               0.0   
1            0.0000                 0.00           0.0  ...               0.0   
2            0.0000                 0.00           0.0  ...               0.0   
3            0.0000                 0.00           0.0  ...               0.0   
4            0.0625                 0.00           0.0  ...               0.0   

   Trail  Train Station  Vegetarian / Vegan Restaurant  Video Game Store  \
0    0.0            0.0                       0.020000               0.0   
1    0.0            0.0                       0.017544               0.0   
2    0.0            0.0                       0.000000               0.0   
3    0.0            0.0                       0.000000               0.0   
4    0.0            0.0                       0.000000               0.0   

   Vietnamese Restaurant  Wine Bar  Wine Shop  Wings Joint  Yoga Studio  
0                    0.0      0.01        0.0          0.0     0.000000  
1                    0.0      0.00        0.0          0.0     0.000000  
2                    0.0      0.00        0.0          0.0     0.000000  
3                    0.0      0.00        0.0          0.0     0.052632  
4                    0.0      0.00        0.0          0.0     0.000000  

[5 rows x 233 columns]

First, let's write a function to sort the venues in descending order.

In [94]:
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

Now let's create the new dataframe and display the top 10 venues for each neighborhood.

In [113]:
num_top_venues = 10

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Neighbourhood']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
neighborhoods_venues_sorted = pd.DataFrame(columns=columns)
neighborhoods_venues_sorted['Neighbourhood'] = toronto_grouped['Neighbourhood']

for ind in np.arange(toronto_grouped.shape[0]):
    neighborhoods_venues_sorted.iloc[ind, 1:] = return_most_common_venues(toronto_grouped.iloc[ind, :], num_top_venues)

neighborhoods_venues_sorted.head()

Neighbourhood 1st Most Common Venue  \
0                           Adelaide, King, Richmond           Coffee Shop   
1                                        Berczy Park           Coffee Shop   
2       Brockton, Exhibition Place, Parkdale Village           Coffee Shop   
3  Business Reply Mail Processing Centre 969 Eastern    Light Rail Station   
4  CN Tower, Bathurst Quay, Island airport, Harbo...       Airport Service   

  2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
0                  Café       Thai Restaurant                   Bar   
1                Bakery          Cocktail Bar              Beer Bar   
2                  Café        Breakfast Spot                   Gym   
3           Yoga Studio                   Spa         Garden Center   
4        Airport Lounge       Harbor / Marina                   Bar   

  5th Most Common Venue  6th Most Common Venue 7th Most Common Venue  \
0            Steakhouse         Cosmetics Shop        Breakfast Spot   
1        Farmers Market             Steakhouse                  Café   
2          Intersection  Performing Arts Venue             Pet Store   
3                Garden   Fast Food Restaurant        Farmers Market   
4                 Plane            Coffee Shop      Sculpture Garden   

  8th Most Common Venue 9th Most Common Venue 10th Most Common Venue  
0            Restaurant                 Hotel           Burger Joint  
1           Cheese Shop    Seafood Restaurant     Italian Restaurant  
2  Gym / Fitness Center          Climbing Gym             Restaurant  
3            Comic Shop                  Park       Recording Studio  
4         Boat or Ferry              Boutique           Airport Gate

## Cluster Neighborhoods
Run k-means to cluster the neighborhood into 5 clusters.

In [112]:
# set number of clusters
kclusters = 3

toronto_grouped_clustering = toronto_grouped.drop('Neighbourhood', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(toronto_grouped_clustering)

# check cluster labels generated for each row in the dataframe
kmeans.labels_[0:10] 

array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1])

Let's create a new dataframe that includes the cluster as well as the top 10 venues for each neighborhood

In [114]:
# add clustering labels
neighborhoods_venues_sorted.insert(0, 'Cluster Labels', kmeans.labels_)

In [115]:


toronto_merged = df_toronto

# merge toronto_grouped with toronto_data to add latitude/longitude for each neighborhood
toronto_merged = toronto_merged.join(neighborhoods_venues_sorted.set_index('Neighbourhood'), on='Neighbourhood')

toronto_merged.head() # check the last columns!

Postcode          Borough                   Neighbourhood   Latitude  \
0      M4E     East Toronto                     The Beaches  43.676357   
1      M4K     East Toronto    The Danforth West, Riverdale  43.679557   
2      M4L     East Toronto  The Beaches West, India Bazaar  43.668999   
3      M4M     East Toronto                 Studio District  43.659526   
4      M4N  Central Toronto                   Lawrence Park  43.728020   

   Longitude  Cluster Labels 1st Most Common Venue 2nd Most Common Venue  \
0 -79.293031               1     Health Food Store  Other Great Outdoors   
1 -79.352188               1      Greek Restaurant           Coffee Shop   
2 -79.315572               1                  Park                   Gym   
3 -79.340923               1                  Café           Coffee Shop   
4 -79.388790               0                  Park           Swim School   

  3rd Most Common Venue 4th Most Common Venue   5th Most Common Venue  \
0                   Pub                 Trail            Neighborhood   
1        Ice Cream Shop    Italian Restaurant  Furniture / Home Store   
2    Italian Restaurant             Pet Store                     Pub   
3           Yoga Studio    Italian Restaurant     American Restaurant   
4              Bus Line           Yoga Studio                 Dog Run   

  6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
0  Ethiopian Restaurant     Electronics Store           Event Space   
1            Restaurant         Grocery Store               Brewery   
2         Movie Theater        Sandwich Place         Burrito Place   
3                Bakery          Neighborhood                 Diner   
4  Fast Food Restaurant        Farmers Market    Falafel Restaurant   

     9th Most Common Venue 10th Most Common Venue  
0       Falafel Restaurant     Dim Sum Restaurant  
1          Bubble Tea Shop   Caribbean Restaurant  
2             Burger Joint                Brewery  
3  Comfort Food Restaurant         Clothing Store  
4              Event Space   Ethiopian Restaurant

In [116]:
# create map
map_clusters = folium.Map(location=[latitude, longitude], zoom_start=11)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(toronto_merged['Latitude'], toronto_merged['Longitude'], toronto_merged['Neighbourhood'], toronto_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

 ### Examine Clusters
 Now, you can examine each cluster and determine the discriminating venue categories that distinguish each cluster. Based on the defining categories, you can then assign a name to each cluster. I will leave this exercise to you.

### Cluster 1

In [117]:
toronto_merged.loc[toronto_merged['Cluster Labels'] == 0, toronto_merged.columns[[2] + list(range(5, toronto_merged.shape[1]))]]

Neighbourhood  Cluster Labels 1st Most Common Venue 2nd Most Common Venue  \
4   Lawrence Park               0                  Park           Swim School   
10       Rosedale               0                  Park            Playground   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
4               Bus Line           Yoga Studio               Dog Run   
10                 Trail    Dim Sum Restaurant        Farmers Market   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
4   Fast Food Restaurant        Farmers Market    Falafel Restaurant   
10    Falafel Restaurant           Event Space  Ethiopian Restaurant   

   9th Most Common Venue       10th Most Common Venue  
4            Event Space         Ethiopian Restaurant  
10     Electronics Store  Eastern European Restaurant

## Cluster 2

In [118]:
toronto_merged.loc[toronto_merged['Cluster Labels'] == 1, toronto_merged.columns[[2] + list(range(5, toronto_merged.shape[1]))]]

Neighbourhood  Cluster Labels  \
0                                         The Beaches               1   
1                        The Danforth West, Riverdale               1   
2                      The Beaches West, India Bazaar               1   
3                                     Studio District               1   
5                                    Davisville North               1   
6                                  North Toronto West               1   
7                                          Davisville               1   
8                         Moore Park, Summerhill East               1   
9   Deer Park, Forest Hill SE, Rathnelly, South Hi...               1   
11                        Cabbagetown, St. James Town               1   
12                               Church and Wellesley               1   
13                                       Harbourfront               1   
14                           Ryerson, Garden District               1   
15                                     St. James Town               1   
16                                        Berczy Park               1   
17                                 Central Bay Street               1   
18                           Adelaide, King, Richmond               1   
19  Harbourfront East, Toronto Islands, Union Station               1   
20           Design Exchange, Toronto Dominion Centre               1   
21                     Commerce Court, Victoria Hotel               1   
23                Forest Hill North, Forest Hill West               1   
24                The Annex, North Midtown, Yorkville               1   
25                     Harbord, University of Toronto               1   
26          Chinatown, Grange Park, Kensington Market               1   
27  CN Tower, Bathurst Quay, Island airport, Harbo...               1   
28                    Stn A PO Boxes 25 The Esplanade               1   
29             First Canadian Place, Underground city               1   
30                                           Christie               1   
31                       Dovercourt Village, Dufferin               1   
32                           Little Portugal, Trinity               1   
33       Brockton, Exhibition Place, Parkdale Village               1   
34                      High Park, The Junction South               1   
35                             Parkdale, Roncesvalles               1   
36                                 Runnymede, Swansea               1   
37  Business Reply Mail Processing Centre 969 Eastern               1   

   1st Most Common Venue 2nd Most Common Venue  3rd Most Common Venue  \
0      Health Food Store  Other Great Outdoors                    Pub   
1       Greek Restaurant           Coffee Shop         Ice Cream Shop   
2                   Park                   Gym     Italian Restaurant   
3                   Café           Coffee Shop            Yoga Studio   
5         Sandwich Place     Food & Drink Shop                    Gym   
6            Coffee Shop   Sporting Goods Shop         Clothing Store   
7         Sandwich Place           Pizza Place           Dessert Shop   
8             Playground                   Gym           Tennis Court   
9                    Pub           Coffee Shop            Pizza Place   
11           Coffee Shop                Bakery     Italian Restaurant   
12           Coffee Shop   Japanese Restaurant       Sushi Restaurant   
13           Coffee Shop                  Park                    Pub   
14           Coffee Shop        Clothing Store                   Café   
15           Coffee Shop                  Café                  Hotel   
16           Coffee Shop                Bakery           Cocktail Bar   
17           Coffee Shop    Italian Restaurant         Ice Cream Shop   
18           Coffee Shop                  Café        Thai Restaurant   
19           Coffee Shop                 Hotel               Aquarium   
20           Coffee Shop

In [119]:
toronto_merged.loc[toronto_merged['Cluster Labels'] == 2, toronto_merged.columns[[2] + list(range(5, toronto_merged.shape[1]))]]

Neighbourhood  Cluster Labels 1st Most Common Venue 2nd Most Common Venue  \
22      Roselawn               2                Garden           Yoga Studio   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
22        Discount Store  Fast Food Restaurant        Farmers Market   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
22    Falafel Restaurant           Event Space  Ethiopian Restaurant   

   9th Most Common Venue       10th Most Common Venue  
22     Electronics Store  Eastern European Restaurant